cmd/server.go

```diff
@@ -67,43 +67,17 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 				EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
 			}
 
-			if storageProvider != "" {
-				snapstoreConfig = &snapstore.Config{
-					Provider:                storageProvider,
-					Container:               storageContainer,
-					Prefix:                  path.Join(storagePrefix, backupFormatVersion),
-					MaxParallelChunkUploads: maxParallelChunkUploads,
-					TempDir:                 snapstoreTempDir,
-				}
-			}
-
-			etcdInitializer := initializer.NewInitializer(options, snapstoreConfig, logger)
-			// Start http handler with Error state and wait till snapshotter is up
-			// and running before setting the state to OK.
-
-			handler := &server.HTTPHandler{
-				Port:            port,
-				EtcdInitializer: *etcdInitializer,
-				Logger:          logger,
-				Status:          http.StatusServiceUnavailable,
-				StopCh:          make(chan struct{}),
-				EnableProfiling: enableProfiling,
-				ReqCh:           make(chan struct{}),
-				AckCh:           make(chan struct{}),
+			if storageProvider == "" {
+				logger.Warnf("No snapstore storage provider configured. Will not start backup schedule. Terminating !!")
```

Comment:
We can't terminate here. We wan't `handler` to be up and running for on demand  initialisation purpose, specifically etcd-events case.

Keep the flow in file same as before. Just create `ssr` object before `handler` start if `storageProvider` is not empty; otherwise pass `nil` to handler. I guess not `handler` have `snapshotter` as well. You can refactor code to avoid `handleNoSsrRequest` call by adding `nil` check in `serverInitialize` function itself.

Reply:
Agreed. I have refactored the code to avoid `handleNoSsrRequest`.